mindscope_utilities/visual_behavior_ophys/data_formatting.py

```diff
@@ -1032,6 +1102,61 @@ def add_time_from_last_change_to_stimulus_presentations(
     return stimulus_presentations
 
 
+def add_n_to_stimulus_presentations(stimulus_presentations):
+    """
+    Adds a column to stimulus_presentations called 'n_after_change',
+    which is the number of stimulus presentations that have occurred since the last change.
+    It will also add a column called 'n_after_omission',
+    which is the number of stimulus presentations that have occurred since the last omission,
+    before the next change.
+    If there is no omission, this value will be -1.
+    Presentations before the first change or omission will have a value of -1.
+    It will also add a column called 'n_before_change',
+    which is the number of stimulus presentations that have occurred before the next change.
+    Presentations after the last change will have a value of -1.
+    Presentations before the first change will also have a value of -1.
+
+    Parameters
+    ----------
+    stimulus_presentations : pd.DataFrame
+        stimulus_presentations table from BehaviorOphysExperiment
+
+    Returns
+    -------
+    stimulus_presentations : pd.DataFrame
+        stimulus_presentations table with 'n_after_change', 'n_after_omission', and 'n_before_change' columns added
+    """
+
+    change_ind = stimulus_presentations[stimulus_presentations['is_change']].index.values
+    n_after_change = np.zeros(len(stimulus_presentations)) - 1  # -1 indicates before the first change
+    for i in range(1, len(change_ind)):
+        n_after_change[change_ind[i - 1]: change_ind[i]] = np.arange(0, change_ind[i] - change_ind[i - 1]).astype(int)
+    n_after_change[change_ind[i]:] = np.arange(0, len(stimulus_presentations) - change_ind[i]).astype(int)
+    stimulus_presentations['n_after_change'] = n_after_change
+    n_before_change = np.zeros(len(stimulus_presentations)) - 1  # -1 indicates after the last and before the first change
+    for i in range(len(change_ind) - 1):
+        n_before_change[change_ind[i] + 1: change_ind[i + 1] + 1] = np.arange(change_ind[i + 1] - change_ind[i] - 1, -1, -1).astype(int)
+    stimulus_presentations['n_before_change'] = n_before_change
+
+    n_after_omission = np.zeros(len(stimulus_presentations)) - 1  # -1 indicates before the first omission or
+    # after the next change till the next omission
```

Comment:
I don't think I understand what "after next change" means, do you mean "before next change"?

Reply:
It was a continuation from the previous comment line.
n_after_omission is defined only up to the next change, so -1 will be assigned from the next change until omission occurs again.